framework/src/modules/interoperability/utils.ts

```diff
@@ -106,3 +146,295 @@ export const computeValidatorsHash = (
+): VerificationResult => {
+	if (!encodedCertificate.equals(EMPTY_BYTES)) {
+		const decodedCertificate = codec.decode<Certificate>(certificateSchema, encodedCertificate);
+		if (
+			decodedCertificate.blockID.equals(EMPTY_BYTES) ||
+			decodedCertificate.stateRoot.equals(EMPTY_BYTES) ||
+			decodedCertificate.validatorsHash.equals(EMPTY_BYTES) ||
+			decodedCertificate.timestamp === 0
+		) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate is missing required values.'),
+			};
+		}
+
+		// Last certificate height should be less than new certificate height
+		if (decodedCertificate.height <= partnerChainAccount.lastCertificate.height) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate height should be greater than last certificate height.'),
+			};
+		}
+	}
+
+	return {
+		status: VerifyStatus.OK,
+	};
+};
+
+export const checkActiveValidatorsUpdate = (
+	txParams: CrossChainUpdateTransactionParams,
+): VerificationResult => {
+	if (
+		txParams.activeValidatorsUpdate.length !== 0 ||
+		txParams.newCertificateThreshold > BigInt(0)
+	) {
+		// Non-empty certificate
+		if (txParams.certificate.equals(EMPTY_BYTES)) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(
+					'Certificate cannot be empty when activeValidatorsUpdate is non-empty or newCertificateThreshold >0.',
+				),
+			};
+		}
+
+		// params.activeValidatorsUpdate has the correct format
+		for (let i = 0; i < txParams.activeValidatorsUpdate.length; i += 1) {
+			const currentValidator = txParams.activeValidatorsUpdate[i];
+			const nextValidator = txParams.activeValidatorsUpdate[i + 1];
+			if (currentValidator.blsKey.byteLength !== VALID_BLS_KEY_LENGTH) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error(`BlsKey length should be equal to ${VALID_BLS_KEY_LENGTH}.`),
+				};
+			}
+			if (currentValidator.blsKey.compare(nextValidator.blsKey) > -1) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Validators blsKeys must be unique and lexicographically ordered'),
+				};
+			}
+		}
+	}
+
+	return {
+		status: VerifyStatus.OK,
+	};
+};
+
+export const checkInboxUpdateValidity = (
+	txParams: CrossChainUpdateTransactionParams,
+	partnerChannelData: ChannelData,
+): VerificationResult => {
+	const partnerChainIDBuffer = getIDAsKeyForStore(txParams.sendingChainID);
+	const decodedCertificate = codec.decode<Certificate>(certificateSchema, txParams.certificate);
+	const { crossChainMessages, messageWitness, outboxRootWitness } = txParams.inboxUpdate;
+	const ccmHashes = crossChainMessages.map(ccm => hash(ccm));
+
+	let newInboxRoot;
+	let newInboxAppendPath = partnerChannelData.inbox.appendPath;
+	let newInboxSize = partnerChannelData.inbox.size;
+	for (const ccm of ccmHashes) {
+		const { appendPath, size } = regularMerkleTree.calculateMerkleRoot({
+			value: ccm,
+			appendPath: newInboxAppendPath,
+			size: newInboxSize,
+		});
+		newInboxAppendPath = appendPath;
```

Comment:
here I think newInboxRoot needs to be assigned as well

Reply:
since this assigned value is not used, and then reassigned later at `newInboxRoot = regularMerkleTree.calculateRootFromRightWitness(
			newInboxSize,
			newInboxAppendPath,
			messageWitness.siblingHashes,
		);`

which is eventually used it throws Deepscan error that assigned value is never used